sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -289,6 +289,13 @@ object AnsiTypeCoercion extends TypeCoercionBase {
           case other => other
         }
         p.makeCopy(Array(a, newList))
+
+      case d @ DateAdd(left @ StringType(), _) if left.foldable =>
```

Comment:
shall we put these 3 in `DateTimeOperations`? The doc of this rule says `Promotes string literals that appear in arithmetic and comparison expressions.`.

Reply:
Well, DateAdd is a sort of arithmetic expression

Reply:
I prefer to put all the string literal promotion here. Let me update the comment.